### PR TITLE
[rust] fix projection in Dataset:take_rows

### PR DESCRIPTION
**Pull request by changhiskhan**

### CODE REVIEW

File: rust/src/dataset/scanner.rs [hunk 1]
```diff
@@ -137,8 +137,15 @@ impl Scanner {
     /// The schema of the output, a.k.a, projection schema.
     pub fn schema(&self) -> SchemaRef {
         if self.nearest.as_ref().is_some() {
+            let q = self.nearest.as_ref().unwrap();
+            let column: ArrowField = self
+                .dataset
+                .schema()
+                .field(q.column.as_str())
+                .unwrap()
```
Comment (eddyxu) [inline rust/src/dataset/scanner.rs:145]: Is this an `Option`? do a `ok_or_else()`?

File: rust/src/io/exec/scan.rs [hunk 2]
```diff
@@ -51,7 +51,7 @@ impl Scan {
     ) -> Self {
         let (tx, rx) = mpsc::channel(prefetch_size);
 
-        let projection = projection.clone();
+        let projection = Arc::new(projection.clone());
```
Comment (eddyxu) [inline rust/src/io/exec/scan.rs:54]: is projection already `Arc<Schema>`? if so, dont need `Arc::new(..)` .

Comment (eddyxu) [reply rust/src/io/exec/scan.rs:54]: Maybe try to run `cargo clippy` to see what reports are there in these changed lines?

**Review by eddyxu — APPROVED** [review]
small nits. lgtm overall.

File: rust/src/dataset/scanner.rs (hunk 1 above)
Comment (changhiskhan) [reply rust/src/dataset/scanner.rs:145]: done. also had to change return type of schema to Result<SchemaRef>

**Pull request merged**